File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         vector<int> v;
         unordered_map<int, int> freq;
         for(int i=0;i<nums.size();i++){
             freq[nums[i]]=i;
         }
         for(int i=0;i<nums.size();i++){
            int x=target-nums[i];
            if(freq.count(x) && freq[x] != i){
                v.push_back(i);
                v.push_back(freq[x]);
                break;
            }
         }
    return v;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
      int n=nums.length;
      HashMap<Integer,Integer> hm=new HashMap<>();
      for(int i=0;i<n;i++){
        hm.put(nums[i],i);
      }  
      int[] a=new int[2];
      for(int i=0;i<n;i++){
        if(hm.containsKey(target-nums[i])){
            if(i!=hm.get(target-nums[i])){
                 a[0]=i;
            a[1]=hm.get(target-nums[i]);
            }
           
        }
      }
      return a;
    }
}